modules/core/src/v2/keychains.ts

```diff
@@ -255,10 +258,31 @@ export class Keychains {
       throw new Error('expected keychain to be an object with an encryptedPrv property');
     }
 
+    if (_.isObject(params.keychain) && _.isString(params.keychain.encryptedPrv) && params.keychain.addressDerivationKeypair) {
+      if (!params.keychain.addressDerivationKeypair.pub) {
+        throw new Error('expected keychain to have addressDerivationKeypair.pub property');
+      }
+      if (!params.keychain.addressDerivationKeypair.encryptedPrv) {
+        throw new Error('expected keychain to have addressDerivationKeypair.encryptedPrv property');
+      }
+    }
```

Comment:
```suggestion
```
remove this, no all keychains have the addressDerivationKeypair so only its necesary to check when they do.